### PR TITLE
ECL-359: Returns - Submitted page updates

### PR DESCRIPTION
**Pull request by nemanjatomanovic**

### CODE REVIEW

File: project/plugins.sbt [hunk 1]
```diff
@@ -2,14 +2,14 @@ resolvers += MavenRepository("HMRC-open-artefacts-maven2", "https://open.artefac
...
+addSbtPlugin("uk.gov.hmrc" % "sbt-distributables" % "2.2.0")
+addSbtPlugin("com.typesafe.play" % "sbt-plugin" % "2.8.18")
+addSbtPlugin("org.scoverage" % "sbt-scoverage" % "1.9.3")
+addSbtPlugin("com.typesafe.sbt" % "sbt-gzip" % "1.0.2")
+addSbtPlugin("io.github.irundaia" % "sbt-sassify" % "1.5.2")
+addSbtPlugin("org.scalastyle" %% "scalastyle-sbt-plugin" % "1.0.0")
+addSbtPlugin("org.scalameta" % "sbt-scalafmt" % "2.4.0")
+addSbtPlugin("net.ground5hark.sbt" % "sbt-concat" % "0.2.0")
+addSbtPlugin("com.typesafe.sbt" % "sbt-uglify" % "2.0.0")
+addSbtPlugin("com.typesafe.sbt" % "sbt-digest" % "1.1.4")
```
Comment (peter-hazell) [inline project/plugins.sbt:15]: Can you add the old formatting back using the scalafmt config, it looks like your intellij formatter has overridden it here?

File: test/uk/gov/hmrc/economiccrimelevyreturns/controllers/ReturnSubmittedControllerSpec.scala [hunk 2]
```diff
@@ -40,28 +40,31 @@ class ReturnSubmittedControllerSpec extends SpecBase {
...
-      )(authRequest, messages).toString
+        contentAsString(result) shouldBe view(
+          chargeReference,
+          ViewUtils.formatToday(),
+          ViewUtils.formatLocalDate(obligationDetails.inboundCorrespondenceDueDate),
+          obligationDetails.inboundCorrespondenceFromDate.getYear.toString,
+          obligationDetails.inboundCorrespondenceToDate.getYear.toString,
+          amountDue
+        )(authRequest, messages).toString()
     }
```
Comment (peter-hazell) [inline test/uk/gov/hmrc/economiccrimelevyreturns/controllers/ReturnSubmittedControllerSpec.scala:69]: Test missing for IllegalStateException when the amount due is not found in session

File: app/uk/gov/hmrc/economiccrimelevyreturns/controllers/CheckYourAnswersController.scala [hunk 3]
```diff
@@ -83,7 +83,14 @@ class CheckYourAnswersController @Inject() (
       request.session
         ++ Seq(
           SessionKeys.ChargeReference   -> response.chargeReference,
-          SessionKeys.ObligationDetails -> Json.toJson(request.eclReturn.obligationDetails).toString()
+          SessionKeys.ObligationDetails -> Json.toJson(request.eclReturn.obligationDetails).toString(),
+          SessionKeys.AmountDue         ->
+            request.eclReturn.calculatedLiability
+              .getOrElse(
+                throw new IllegalStateException("Amount due was not found.")
```
[the remark this answers is not in the document]
Comment (nemanjatomanovic) [reply app/uk/gov/hmrc/economiccrimelevyreturns/controllers/CheckYourAnswersController.scala]: Done.

File: project/plugins.sbt (hunk 1 above)
Comment (nemanjatomanovic) [reply project/plugins.sbt]: Nice catch, thanks.

File: app/uk/gov/hmrc/economiccrimelevyreturns/controllers/ReturnSubmittedController.scala [hunk 4]
```diff
@@ -51,7 +51,7 @@ class ReturnSubmittedController @Inject() (
     val amountDue: BigDecimal = BigDecimal(
       request.session
         .get(SessionKeys.AmountDue)
-        .getOrElse(throw new IllegalStateException("Amount due not found in session"))
+        .getOrElse(throw new IllegalStateException("Amount due not found in return data"))
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/economiccrimelevyreturns/controllers/ReturnSubmittedController.scala:54]: `Amount due not found in session` was fine, as this is pulling data from the session :)

File: project/plugins.sbt [hunk 5]
```diff
@@ -2,14 +2,14 @@ resolvers += MavenRepository("HMRC-open-artefacts-maven2", "https://open.artefac
...
-addSbtPlugin("org.scalastyle"           %% "scalastyle-sbt-plugin" % "1.0.0")
-addSbtPlugin("org.scalameta"            % "sbt-scalafmt"           % "2.4.0")
-addSbtPlugin("net.ground5hark.sbt"      % "sbt-concat"             % "0.2.0")
-addSbtPlugin("com.typesafe.sbt"         % "sbt-uglify"             % "2.0.0")
-addSbtPlugin("com.typesafe.sbt"         % "sbt-digest"             % "1.1.4")
+addSbtPlugin("org.scalastyle" %% "scalastyle-sbt-plugin" % "1.0.0")
+addSbtPlugin("org.scalameta" % "sbt-scalafmt" % "2.4.0")
+addSbtPlugin("net.ground5hark.sbt" % "sbt-concat" % "0.2.0")
+addSbtPlugin("com.typesafe.sbt" % "sbt-uglify" % "2.0.0")
+addSbtPlugin("com.typesafe.sbt" % "sbt-digest" % "1.1.4")
```
Comment (peter-hazell) [inline project/plugins.sbt:15]: I think you have only formatted one line ?

File: app/uk/gov/hmrc/economiccrimelevyreturns/controllers/ReturnSubmittedController.scala (hunk 4 above)
Comment (peter-hazell) [reply app/uk/gov/hmrc/economiccrimelevyreturns/controllers/ReturnSubmittedController.scala:54]: sure - difference here is that you are pulling data from the session, in the check your answers controller you are pulling data from mongo - so they are distinctive 😃

File: project/plugins.sbt (hunk 5 above)
Comment (nemanjatomanovic) [reply project/plugins.sbt]: Something weird happened, sorry about that, but should be fine now.

**Review by peter-hazell — APPROVED** [review]

**Pull request merged**